lib/lhm/throttler/slave_lag.rb

```diff
@@ -75,9 +91,19 @@ def slave_lag(slave)
 
       def slave_connection(slave)
         adapter_method = defined?(Mysql2) ? 'mysql2_connection' : 'mysql_connection'
+        begin
+          ActiveRecord::Base.send(adapter_method, slave_config(slave))
+        rescue Mysql2::Error => e
+          Lhm.logger.info "Error conecting to #{slave}: #{e}"
+          nil
+        end
+      end
+
+      def slave_config(slave)
         config = ActiveRecord::Base.connection_pool.spec.config.dup
+        config = config[@get_current_shard.call.to_sym] if @get_current_shard && @get_current_shard.call
```

Comment:
Is this assuming that the `config` hash has nested configurations of each shard? This seems like it's pretty specific.